src/others/sysmon/lv_sysmon.c

```diff
@@ -15,16 +15,15 @@
 #include "../../misc/lv_async.h"
 #include "../../stdlib/lv_string.h"
 #include "../../widgets/label/lv_label.h"
+#include "../../display/lv_display_private.h"
 
 /*********************
  *      DEFINES
  *********************/
-#define MY_CLASS (&lv_sysmon_class)
 
 #define SYSMON_REFR_PERIOD_DEF 300 /* ms */
 
 #if defined(LV_USE_PERF_MONITOR) && LV_USE_PERF_MONITOR
-    #define sysmon_perf LV_GLOBAL_DEFAULT()->sysmon_perf
     #define _USE_PERF_MONITOR   1
 #else
     #define _USE_PERF_MONITOR   0
```

Comment:
Is it possible to modify `lvgl/scripts/lv_conf_internal_gen.py`,  to always define LV_USE_PERF_MONITOR even if `LV_USE_SYSMON` is 0? 

`LVGL`'s style is to check `#if LV_USE_PERF_MONITOR`, so it'd better define it from the first place.
```c
/*1: Enable system monitor component*/
#ifndef LV_USE_SYSMON
    #ifdef CONFIG_LV_USE_SYSMON
        #define LV_USE_SYSMON CONFIG_LV_USE_SYSMON
    #else
        #define LV_USE_SYSMON   0
    #endif
#endif

//////////////#if LV_USE_SYSMON
    /*1: Show the used memory and the memory fragmentation
     * Requires `LV_USE_STDLIB_MALLOC = LV_STDLIB_BUILTIN`
     * Requires `LV_USE_SYSMON = 1`*/
    #ifndef LV_USE_MEM_MONITOR
        #ifdef CONFIG_LV_USE_MEM_MONITOR
            #define LV_USE_MEM_MONITOR CONFIG_LV_USE_MEM_MONITOR
        #else
            #define LV_USE_MEM_MONITOR 0
        #endif
    #endif
    #if LV_USE_MEM_MONITOR
        #ifndef LV_USE_MEM_MONITOR_POS
            #ifdef CONFIG_LV_USE_MEM_MONITOR_POS
                #define LV_USE_MEM_MONITOR_POS CONFIG_LV_USE_MEM_MONITOR_POS
            #else
                #define LV_USE_MEM_MONITOR_POS LV_ALIGN_BOTTOM_LEFT
            #endif
        #endif
////////////#endif
```

Reply:
having `LV_USE_SYSMON` in the `lv_conf.h` file is pretty useless and it should not be something the user sets. It's a simple test to see if either `LV_USE_MEM_MONITOR`  or `LV_USE_PERF_MONITOR` is set in order to set `LV_USE_SYSMON ` accordingly.


Something along the lines of..


```c
#ifndef LV_USE_SYSMON
    #if LV_USE_MEM_MONITOR || LV_USE_PERF_MONITOR
        #define LV_USE_SYSMON 1
    #else
        #define LV_USE_SYSMON 0
    #endif
#endif
```

Reply:
I had also thought that when using the macros if the macro is not defined then doing the following would evaluate as `false`. The only time when testing to see if a macro is defined or not is to be used as a guard against setting it a second time which results in a compiler warning or error.

```c
#if LV_USE_PERF_MONITOR
```

Reply:
As @FASTSHIFT pointed out `sysmon` can be used by the user too, so there should be a way to enable it independently.

I've update `lv_conf_internal_gen.py`.